UI/Victor_WebStore/Controllers/CatalogController.cs

```diff
@@ -24,6 +24,8 @@ public IActionResult Index()
         public IActionResult ProductDetails(int id)
         {
             var product = _productService.GetProductById(id);
+            if (product is null)
+               return RedirectToActionPermanent("NotFound404", "home");
```

Comment:
```suggestion
               return RedirectToAction("NotFound404", "home");
```

Нужны очень веские обоснования применения Permanent. Браузер кеширует этот ответ и всегда будет автоматически переходить по указанному адресу. Даже если Вы измените поведение этого метода, то браузер об этом уже никогда не узнает.